stats/stats.go

```diff
@@ -0,0 +1,74 @@
+package stats
+
+import (
+	"sync"
+	"time"
+)
+
+type Stats struct {
+	// todo atomic instead of lock?
+	lock *sync.Mutex
```

Comment:
Not sure that it's possible to use atomic for maps.
But RWMutex will be suitable there.

Reply:
RWMutex seems more convenient, but I'm not convinced we need it here.
RWMutex would come in handy here if we dealt with multiple readers and a few writers. Also using RWMutex comes with a performance hit.

More meaningful optimization would be that we give up on those maps and introduce fields for each message type in the Stats itself, and then use atomics.

But for now, I think we're ok

Reply:
I thought a bit and realized that it's better to use Mutex here. It can guarantee that stats is consistent.